kong/plugins/response-transformer/header_transformer.lua

```diff
@@ -63,6 +63,15 @@ function _M.transform_headers(conf, headers)
       kong.response.clear_header(header_name)
   end
 
+  -- Rename headers(s)
+  for _, old_name, new_name in iter(conf.rename.headers) do
+    if headers[old_name] ~= nil and new_name then
+      local value = headers[old_name]
+      kong.response.set_header(new_name, value)
```

Comment:
should this be `add_header` to cover the case where a header supports multiple entries?

e.g.

```
A: 1
B: 2

"rename": "B:A"

A: 1
A: 2
```

The "overwriting rename" could be achieved with

```
A: 1
B: 2

"remove": "A",  "rename": "B:A"

A: 2
```

How does that sound like?...

Reply:
I think we have to get the same behaviour of request-transformer plugin to avoid surprise for developers.

In the request-transformer plugin then rename function does:

```
req_set_header(new_name, value) // Set the current request's request header named header_name to value header_value, overriding any existing ones.
req_clear_header(old_name)
```

In the future we can review "rename" behaviour, but now I think we must keep both consistently.

Reply:
@aalmazanarbs you're completely right, I forgot that request-transformer already had a `rename` feature! Agreed about consistency.